### PR TITLE
Add mutated forest creatures.

### PR DESCRIPTION
**Pull request by Kanexan**
#### Summary
SUMMARY: Content "Adds a selection of mutated woodland creatures"

#### Purpose of change
The purpose of the change is to add additional variation and color into the game's wildlife, and to demonstrate non-zombie Blob mutation on terrestrial fauna. 

#### Describe the solution
This PR adds mutated versions of the deer, wolf, dog, cat, and bear, plus two separate mutated versions of the beaver and the coyote. It also adds them into the relevant spawngroups, and creates harvest.json entries for small, normal, and large mutant animals with fur.

#### Describe alternatives you've considered
Not creating these animals. I also considered making it so that the "base" animals had chances over time to mutate into these, but decided to leave that feature out of this PR for now; if that becomes a part, it would likely be a good idea to make more versions of each animal so that there's the appropriate amount of variation from mutation.

#### Testing
Loaded up a world and made sure that all animals could spawn without causing errors, and gave the expected behavior and butchery results.

#### Additional context
Screenshot (116)
This is a screenshot of all the mutant animals, loaded in-game.

### CODE REVIEW

**Comment by dissociativity** [comment]
can you give these a looks-like for the closest creatures appearance/statwise?

**Comment by Kanexan** [comment]
Will do, as soon as I have the time; shouldn't be too long. I'm not entirely sure what in mutant_monsters.json isn't linting, since it seems to pass the web linter as far as I can tell, but it's completely unsurprising to me that it's taking issue with wilderness.json.

**Comment by Kanexan** [comment]
It turns out that the JSON requires two spaces after a period. Everything _should_ be linting correctly now, but it's asking me to change the formatting in wilderness.json to a different format which I was under the belief was no longer correct. About half of wilderness.json uses the format I used, which is how the online linter styles it, and about half uses the format Clang wants.

**Comment by Venera3** [comment]
The flavor on these is adorable, and I'm now sad that a beaver birb is not a thing in real life.
In the future you could add both these and the zombie animals to the evolution tree of normal critters, to show that either they got deaded and rose as zeds or they thrived in the postapocalypse and got *weird* doing so (like the player usually does). 

On the combat front, both the wolf and the bear look a bit undertuned for late game where players can run around with 10+ cut and bash armor without difficulty. You might want to add some armor piercing or raw damage to both. The bear is also a bit too fast, also making it attack faster (speed is used for attack speed too). If the heads are functional, they could also get "STUMBLES" to show them getting distracted and to make outrunning them possible when lucky.

File: data/json/monsters/mutant_mammal.json [hunk 1]
```diff
@@ -0,0 +1,390 @@
+[
+  {
+    "id": "mon_bear_mutant",
```
Comment (I-am-Erk) [inline data/json/monsters/mutant_mammal.json:3]:
```suggestion
    "id": "mon_bear_mutant_2headed",
```
I would suggest that all of these IDs be made more precise... I have a lot of plans for other mutant animals, and when there's more variety it'll be annoying to find the right IDs if we start out vague.

**Review by I-am-Erk — DISMISSED** [review]
Good stuff, I've been wanting to get on mutant animals for a bit and it's always nice when someone else gets the ball rolling.

Any of these that would be common lab test animals (dogs and cats at least) could be included in lab monster spawn lists.

**Comment by Kanexan** [comment]
Made the IDs more precise and edited the bear and wolf somewhat; they now both have biting attacks, and I added the STUMBLE flag (and a small reduction in speed from the baseline bear) to the bear. Thanks for the feedback!

**Comment by AMurkin** [comment]
> It turns out that the JSON requires two spaces after a period. Everything _should_ be linting correctly now, but it's asking me to change the formatting in wilderness.json to a different format which I was under the belief was no longer correct. About half of wilderness.json uses the format I used, which is how the online linter styles it, and about half uses the format Clang wants.

If you are talking about line breaks you need to format JSON as "JSON style check" requires it. Just format the entire file in the [web linter](http://dev.narc.ro/cataclysm/format.html). The format depends on the length of the string.
```
-      {
-        "monster": "mon_bear_mutant_3headed",
-        "freq": 2,
-        "cost_multiplier": 10,
-        "starts": 672
-      },
+      { "monster": "mon_bear_mutant_3headed", "freq": 1, "cost_multiplier": 10 },
```

**Comment by Venera3** [comment]
Do you mind if I take them out for a spot of combat testing? I'm kind of twiddling my thumbs till my dragonfly stuff merges, and set up a nice combat test environment for those.

**Comment by Kanexan** [comment]
I wouldn't mind at all! Feedback on their respective combat prowess as things stand would be _much_ appreciated; I mostly fiddled around with numbers until it seemed appropriately deadly when loaded into the game, so having outside feedback on that would be invaluable.

**Comment by aoae23** [comment]
Pleasantly surprised to see the spider deer edit get implemented into the game.

Glancing at their stats, their melee skills seem balanced at least at first glance.

**Comment by Venera3** [comment]
So I spent a bit bothering the weird wildlife (well, the more capable bunch). Since they start spawning pretty late I skipped the day1 survivors and ran early/midgame ( 3/3/3 melee-bash-dodge, quarterstaff, no armor) and proper lategame (++survivor gear+MBR, nodachi, 7/7/7 skills). Everybody got 10 tries, Brawling+TAB, no funny "tactics" or "guns". 

Results:
Cerbearuses (Cerbeari?) get chased away 10/10 by a midgame char, get mostly 2-3 hits in before bugging out. They also can't put a dent in lategame armor.
Dire wolves fare slighty better, they still get chased off by the midgame char but at least get a few limbs down to red/broken. In the late game they can only bite through the non-MBR armored extremities, 1v1 they usually get one hit in, leaving everything in the green.
Venom coyotes were the only critters capable of ganking a midgame char 8v1 (their theoretical max pack size) once, since venom is a pretty debilitating debuff and makes running away a chore. If they don't get some good hits in, however the hit and run AI screws them over big time and they can't do anything to a lategame toon.
Shark coyotes are most the same, without the AI quirks they could threaten a midgame character in packs, but not as they stand.

Suggestions:
Generally, everyone could use higher damage both as a baseline and for their special attacks. Don't forget that the player has effectively 500 health + (outside of the proper early game) some form of armor. If they stay as lategame threats their melee skill could be bumped up too, since they'll have to deal with a player who leveled dodge. Armor_multiplier is a weird variable I personally could not get to work consistently, armor_penetration is much more straightforward for the special attacks. Apply with gusto.
The bears could use more hp, since morale is dependant on the remaining health, and they run away very early. Also bears are chunky boys.
The coyotes are definitely hampered by the hit-and-run AI, other than that + generally lacking damage they feel fine. With the low morale you might want to add GROUP_MORALE to them so that they stay a bit longer.
The direwolf stands to be the proper Bad News critter of this bunch, so I'd pull out all the stops for them damage and skill-wise. I'd also add a bit of hp, but that's secondary.

As an aside, I'm fairly certain placate:MEAT does nothing, but I'd be happy to be corrected on that one.

**Review by I-am-Erk — COMMENTED** [review]
I think you could probably give the three headed bear a bit more armour and the extra good smell trait, but basically these look done to me. I'll check over once more in the morning and possibly merge then.

**Comment by l29ah** [comment]
A grinning coyote got my freshly spawned character, easily finding, outrunning and killing it in a few hits. Is it really its intended performance, being a death sentence at day0?

**Comment by Kanexan** [comment]
At day zero? Yeah, that specific critter is going to be pretty lethal in melee combat, especially in a pack. They're very, very infrequent spawns, however.

**Comment by l29ah** [comment]
> At day zero?

Yes, i spawned with Last Flight scenario and died after 18 minutes to a grinning coyote, missing three times with my stick while it ravaged through my body.

**Pull request merged**